src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,93 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { getPostComments, deletePostComments } from '../../api/comments';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+interface Props {
+  postDetails: Post,
+}
+
+export const PostDetails: React.FC<Props> = ({ postDetails }) => {
+  const postId = postDetails.id;
+  const postBody = postDetails.body;
+
+  const [comments, setComments] = useState([]);
+  const [show, setShow] = useState(true);
+
+  const requestGetPostComments = (id: number) => {
+    getPostComments(id)
+      .then(newComments => {
+        setComments(newComments);
+      });
+  };
+
+  useEffect(() => {
+    requestGetPostComments(postId);
+  }, [postId]);
+
+  const handleSetShow = () => {
+    setShow(!show);
```

Comment:
```suggestion
    setShow((prev) => !prev);
```
Используй колбек, чтобы поменять стейт на основании предыдущего значения